lib/api/BackbeatAPI.js

```diff
@@ -615,6 +622,150 @@ class BackbeatAPI {
         err => cb(err, response));
     }
 
+    /**
+     * Tempfix to correct failure/pending metrics on concurrency issue
+     * @param {array} keys - an array of redis failure keys to remove.
+     *   Schema:
+     *   bb:crr:failed:<bucket>:<key>:<versionID>:<storageClass>:<timestamp>
+     * @return {undefined}
+     */
+    _normalizeFailMetrics(keys) {
+        // ignore deleting duplicate keys for tests
+        if (process.env.CI === 'true') {
+            return;
+        }
+        async.eachLimit(keys, 10, (key, next) => {
+            async.waterfall([
+                done => this._redisClient.batch([['get', key]], (err, res) => {
```

Comment:
Since we are not targeting efficiency I think it's fine as it is, but we could also have used a single batch to get all keys.